File: HealthSysHub.Web.DBConfiguration/Models/PharmacyPayment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HealthSysHub.Web.DBConfiguration.Models
{
    [Table("PharmacyPayment")]
    public class PharmacyPayment
    {
        [Key]
        public Guid PaymentId { get; set; }
        public Guid? PharmacyOrderId { get; set; }
        public Guid? HospitalId { get; set; }
        public Guid? PharmacyId { get; set; }
        public string? PaymentNumber { get; set; } // NOT NULL UNIQUE
        public DateTimeOffset? PaymentDate { get; set; } = DateTimeOffset.Now;
        public string? PaymentMethod { get; set; } // NOT NULL, must be one of the specified values
        public decimal? PaymentAmount { get; set; } // NOT NULL
        public string? ReferenceNumber { get; set; } // Transaction ID, check number, etc.
        public string? Status { get; set; } // NOT NULL, must be one of the specified values
        public string? PaymentGateway { get; set; } // For online payments
        public string? GatewayResponse { get; set; } // Raw gateway response
        public string? Notes { get; set; }
        public Guid? CreatedBy { get; set; }
        public DateTimeOffset CreatedOn { get; set; } = DateTimeOffset.Now;
        public Guid? ModifiedBy { get; set; }
        public DateTimeOffset? ModifiedOn { get; set; }
        public bool IsActive { get; set; } = true;
    }

}

File: HealthSysHub.Web.Pharmacy.UI/src/app/models/pharmacy-payment.ts
export interface PharmacyPayment {
    paymentId?: string; // Guid in C# is represented as a string in TypeScript
    pharmacyOrderId?: string; // Optional Guid
    hospitalId?: string; // Optional Guid
    pharmacyId?: string; // Optional Guid
    paymentNumber: string; // NOT NULL UNIQUE
    paymentDate?: string; // DateTimeOffset as string
    paymentMethod: string; // NOT NULL
    paymentAmount: number; // NOT NULL
    referenceNumber?: string; // Optional
    status: string; // NOT NULL
    paymentGateway?: string; // Optional
    gatewayResponse?: string; // Optional
    notes?: string; // Optional
    createdBy?: string; // Optional Guid
    createdOn: string; // DateTimeOffset as string
    modifiedBy?: string; // Optional Guid
    modifiedOn?: string; // Optional DateTimeOffset as string
    isActive: boolean; // Default is true
}
